src/renderers/dom/fiber/wrappers/ReactDOMFiberSelect.js

```diff
@@ -103,6 +107,9 @@ function updateOptions(
     for (let i = 0; i < options.length; i++) {
       if (options[i].value === selectedValue) {
         options[i].selected = true;
+        if (setDefaultSelected) {
+          options[i].defaultSelected = true;
```

Comment:
perhaps this should be set once on mount? That is a bit iffy i guess. Ostensibly `defaultValue` is set once for inputs and changing it after that is a noop for the life of the component. It seems like now tho you can set it, change it and reset the form and it'll use the new value. That may not be worth the complexity to address, but it opens up a new inconsistency perhaps.

Reply:
> Ostensibly defaultValue is set once for inputs and changing it after that is a noop for the life of the component.

That isn't the case, as far as I've seen. [Here's an example](https://jsfiddle.net/gocubxhk/) that you can update `defaultValue` and a form reset will reset to the new `defaultValue` option. That happens in `ReactDOMFiberInput.updateWrapper` [here](https://github.com/facebook/react/blob/master/src/renderers/dom/fiber/wrappers/ReactDOMFiberInput.js#L220-L232). In both cases, updating `.defaultValue` is guarded by a loose `null` check for `value` first, so I think it should be consistent?

Reply:
ah ok, so it's consistent with the other input reset behavior sorry. carry on then :P